codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/EndpointResolverMiddleware.kt

```diff
@@ -53,14 +53,9 @@ class EndpointResolverMiddleware(
         writer.write("let endpoint = try endpointResolver.resolve(params: endpointParams)")
             .write("")
 
-        writer.openBlock("""guard let authScheme = endpoint.authScheme(name: "sigv4") else {""", "}") {
```

Comment:
Any idea what the rationale was for enforcing the auth scheme in the first place?

Reply:
That was a bad assumption that we will always a `authscheme` present.

In past also, both properties inside the authScheme were optional https://github.com/awslabs/aws-sdk-swift/blob/0.3.1/AWSClientRuntime/Sources/Middlewares/EndpointResolverMiddleware.swift#L49-L55